scripts/apps/search/MultiImageEdit.ts

```diff
@@ -161,23 +162,34 @@ export function MultiImageEditController(
 
     }
 
+    function getUniqueValuesExtra(field: string) {
+        const uniqueValues = {};
+        $scope.getSelectedImages()
+            .filter((item) => item.extra != null && item.extra[field] != null && item.extra[field] !== '')
+            .map((item) => item.extra[field])
+            .forEach((value) => uniqueValues[value] = 1);
+        return Object.keys(uniqueValues);
+    }
+
+    /**
+     * Populate .extra metadata for editing.
+     *
+     * Works like compare() but for custom fields stored in .extra.
```

Comment:
Could the same code be reused between `compareExtra` and `compare` with a different argument specifying which part of the object to check?